root/components/RelationshipsTable.js

```diff
@@ -0,0 +1,177 @@
+    if (hasCreditColumn && hasAttributeColumn &&
+      hasArtistColumn && hasLengthColumn) {
+      break;
+    }
+  }
+
+  const columnsCount = 1 + hasCreditColumn +
+                       hasAttributeColumn + hasArtistColumn + hasLengthColumn;
+  return (
+    <>
+      <h2>{heading}</h2>
+      <table className="tbl">
+        <thead>
+          <tr>
+            <th>{l('Date')}</th>
+            <th>{l('Title')}</th>
+            {hasCreditColumn ? <th>{l('Credited As')}</th> : null}
+            {hasAttributeColumn ? <th>{l('Attributes')}</th> : null}
+            {hasArtistColumn ? <th>{l('Artist')}</th> : null}
+            {hasLengthColumn ? <th>{l('Length')}</th> : null}
+          </tr>
+        </thead>
+        <tbody>
+          {Object.keys(appearances).sort().map(relationshipType => (
+            <React.Fragment key={relationshipType}>
+              <tr className="subh">
+                <th />
+                <th colSpan={columnsCount}>
+                  {l_relationships(relationshipType)}
+                </th>
+              </tr>
+              {appearances[relationshipType].map((relationship, index) => {
+                const attributes = relationship.attributes;
+                const attributeList = [];
+                let sourceCredit = '';
+                let targetCredit = '';
+                if (relationship.direction === 'backward') {
+                  targetCredit = relationship.entity0_credit;
+                  sourceCredit = relationship.entity1_credit;
+                } else {
+                  sourceCredit = relationship.entity0_credit;
+                  targetCredit = relationship.entity1_credit;
+                }
+
+                return (
+                  <tr className={loopParity(index)} key={relationship.id}>
+                    <td>{formatDatePeriod(relationship)}</td>
+                    <td>
+                      {relationship.editsPending ? (
+                        <span className="mp mp-rel">
+                          <EntityLink
+                            content={targetCredit}
+                            entity={relationship.target}
+                          />
+                        </span>
+                      ) : (
+                        <EntityLink
+                          content={targetCredit}
+                          entity={relationship.target}
+                        />
+                      )}
+                    </td>
+                    {hasCreditColumn ? (
+                      <td>
+                        {sourceCredit || null}
+                      </td>
+                    ) : null}
+                    {hasAttributeColumn ? (
+                      <td>
+                        {attributes && attributes.length > 0 ? (
+                          attributes.forEach(attribute => {
+                            attributeList.push(displayLinkAttribute(attribute));
+                          }),
+                          commaList(attributeList)
```

Comment:
You can replace the `forEach` with `map` here, and pass it directly to `commaList`. Then `attributeList` can be removed, too